[osgEarth] Update port to osgEarth 3.7

plevy opened the pull request:
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ x] SHA512s are updated for each updated download.
- [ x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ x] Any patches that are no longer applied are deleted from the port's directory.
- [ x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ x] Only one version is added to each modified port's versions file.

plevy committed:
draft osgEarth 3.6.1 port

plevy committed:
update versions

plevy committed:
version should be 3.7

plevy commented:
This previous version of the port was osgEarth 3.4 and in preparation for 3.7, we incorporated or fixed most of the previous patches into the new cmake configuration.  Thank you

plevy committed:
updated versions

dg0yt commented on ports/osgearth/portfile.cmake:
```diff
@@ -1,17 +1,12 @@
     REPO gwaldron/osgearth
-    REF "osgearth-${VERSION}"
-    SHA512 f65c31922bebcbf722474a047dc29c8c1ceec9c037b0704811af2627fc2d0a124b6e95888e7d3b9b0e5acc146a88ebf8669e3f864a75a91751c3a4571d05a630
+    REF 1a4344609c0e58aa3e6b1a9cf7a6f82984337e6c
```
With vcpkg.json updated, you can restore
```suggestion
    REF "osgearth-${VERSION}"
```

dg0yt replied:
I see that osgearth 3.7 isn't release yet. Then this PR should be a draft until the git tag is available and immutable.

dg0yt commented on ports/osgearth/portfile.cmake:
```diff
@@ -63,6 +48,7 @@ vcpkg_cmake_configure(
         -DOSGEARTH_BUILD_PROCEDURAL_NODEKIT=OFF
         -DOSGEARTH_BUILD_TRITON_NODEKIT=OFF
         -DOSGEARTH_BUILD_SILVERLINING_NODEKIT=OFF
+		-DOSGEARTH_BUILD_ZIP_PLUGIN=OFF		
```
Why this turned off now?

dg0yt commented on ports/osgearth/portfile.cmake:
```diff
@@ -72,7 +58,8 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
-vcpkg_cmake_config_fixup(CONFIG_PATH cmake/)
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME osgEarth)
```
```suggestion
vcpkg_cmake_config_fixup()
```
No extra blankline. No explicit PACKAGE_NAME. (It is only used for the directory name, and the default `osgearth` is fine for vcpkg and CMake.)

dg0yt commented on versions/o-/osgearth.json:
```diff
@@ -1,5 +1,15 @@
+      "version": "3.7",
+      "port-version": 0
+    },
+    {
+      "git-tree": "7298312c829d84b84a8052829b064622d8ca67d5",
+      "version": "3.6.1",
+      "port-version": 0
+    },
```
```suggestion
```
One PR can add only one entry. `"git-tree": "7298312c829d84b84a8052829b064622d8ca67d5"` won't exist after the merge.

LilyWangLL replied:
Pinging @plevy for response, can you address the review suggestions? Thanks!

plevy replied:
Sorry, missed this but committed the suggestion.  Checks are running but taking forever, maybe something is hung.  I don't want to commit something just to run the checks again

dg0yt replied:
> Checks are running but taking forever, maybe something is hung.

Some recent merges invalidated nearly all cached artifacts. So it takes much longer to arrive at the actual changes.

plevy committed:
update to latest 3.7 tag

plevy committed:
update versions

plevy committed:
remove extra line

plevy committed:
update versions

plevy commented:
I think I addressed all the issues

- We tagged a 3.7 release to use  REF "osgearth-${VERSION}"
- Fixed the install share folder naming problem and removed the package name from vcpkg_cmake_config_fixup()
- I believe the options that are turned OFF are the same as the prior portfile.

Thank you

dg0yt commented:
> * I believe the options that are turned OFF are the same as the prior portfile.

`OSGEARTH_BUILD_ZIP_PLUGIN` is explicitly turned off by this PR. I didn't check the project's default, but I guess that the `libzip` dependency exists for this plugin.

dg0yt commented:
Is `libzip` still needed as a direct dependency with the plugin turned off?

JavierMatosD commented:
@plevy it doesn't look like upstream has a 3.7 release. I'm marking this draft until upstream mints a release

plevy committed:
remove libzip as not needed

plevy committed:
update versions

plevy commented:
We added the release here:
https://github.com/gwaldron/osgearth/releases/tag/osgearth-3.7
and I removed libzip from vcpkg.json

Thank you

plevy committed:
Update versions/o-/osgearth.json

3.6.1 version was previous version but we moved to 3.7

Co-authored-by: Kai Pastor <dg0yt@darc.de>

LilyWangLL commented:
All features passed with following triplets:
```
x64-windows
x64-windows-static
```
Usage test passed on `x64-windows`.

LilyWangLL commented:
There are the following errors when installing `osgearth[tools]:x64-windows-static`, could you please resolve these errors?

```
[458/476] C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=src\applications\osgearth_viewer\CMakeFiles\osgearth_viewer.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\osgearth_viewer.rsp  /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBLINK: command "C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe @CMakeFiles\osgearth_viewer.rsp /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF /subsystem:console /MANIFEST:EMBED,ID=1" failed (exit code 1120) with the following output:
   Creating library src\applications\osgearth_viewer\osgearth_viewer.lib and object src\applications\osgearth_viewer\osgearth_viewer.exp
LINK : warning LNK4217: symbol '?initialize@osgEarth@@YAXAEAVArgumentParser@osg@@@Z (void __cdecl osgEarth::initialize(class osg::ArgumentParser &))' defined in 'osgEarth.lib(Registry.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
LINK : warning LNK4217: symbol '??0UnitsType@osgEarth@@QEAA@PEBD0AEBW4Domain@Units@1@N@Z (public: __cdecl osgEarth::UnitsType::UnitsType(char const *,char const *,enum osgEarth::Units::Domain const &,double))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic initializer for 'BAM''(void)" (??__EBAM@Units@osgEarth@@YAXXZ)'
LINK : warning LNK4217: symbol '??0UnitsType@osgEarth@@QEAA@PEBD0AEBV01@1@Z (public: __cdecl osgEarth::UnitsType::UnitsType(char const *,char const *,class osgEarth::UnitsType const &,class osgEarth::UnitsType const &))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic initializer for 'DATA_MILES_PER_HOUR''(void)" (??__EDATA_MILES_PER_HOUR@Units@osgEarth@@YAXXZ)'
LINK : warning LNK4217: symbol '??1UnitsType@osgEarth@@QEAA@XZ (public: __cdecl osgEarth::UnitsType::~UnitsType(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic atexit destructor for 'BAM''(void)" (??__FBAM@Units@osgEarth@@YAXXZ)'
LINK : warning LNK4217: symbol '??0ShaderGenerator@osgEarth@@QEAA@XZ (public: __cdecl osgEarth::ShaderGenerator::ShaderGenerator(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
LINK : warning LNK4217: symbol '?run@ShaderGenerator@osgEarth@@QEAAXPEAVNode@osg@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAVStateSetCache@2@@Z (public: void __cdecl osgEarth::ShaderGenerator::run(class osg::Node *,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class osgEarth::StateSetCache *))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
LINK : warning LNK4217: symbol '??_DShaderGenerator@osgEarth@@QEAAXXZ (public: void __cdecl osgEarth::ShaderGenerator::`vbase destructor'(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static class osgEarth::MapNode * __cdecl osgEarth::MapNode::get(class osg::ref_ptr<class osg::Node> &,unsigned int)" (__imp_?get@MapNode@osgEarth@@SAPEAV12@AEAV?$ref_ptr@VNode@osg@@@osg@@I@Z) referenced in function main
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl osgEarth::Util::EarthManipulator::EarthManipulator(class osg::ArgumentParser &)" (__imp_??0EarthManipulator@Util@osgEarth@@QEAA@AEAVArgumentParser@osg@@@Z) referenced in function main
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual char const * __cdecl osgEarth::Util::EarthManipulator::className(void)const " (__imp_?className@EarthManipulator@Util@osgEarth@@UEBAPEBDXZ) referenced in function "[thunk]:public: virtual char const * __cdecl osgEarth::Util::EarthManipulator::className`vtordisp{4294967292,0}' (void)const " (?className@EarthManipulator@Util@osgEarth@@$4PPPPPPPM@A@EBAPEBDXZ)
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) protected: void __cdecl osgEarth::Util::EarthManipulator::`vbase destructor'(void)" (__imp_??_DEarthManipulator@Util@osgEarth@@IEAAXXZ) referenced in function "protected: virtual void * __cdecl osgEarth::Util::EarthManipulator::`scalar deleting destructor'(unsigned int)" (??_GEarthManipulator@Util@osgEarth@@MEAAPEAXI@Z)
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class osg::ref_ptr<class osg::Node> __cdecl osgEarth::Util::MapNodeHelper::load(class osg::ArgumentParser &,class osgViewer::ViewerBase *)const " (__imp_?load@MapNodeHelper@Util@osgEarth@@QEBA?AV?$ref_ptr@VNode@osg@@@osg@@AEAVArgumentParser@5@PEAVViewerBase@osgViewer@@@Z) referenced in function main
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgEarth::Util::MapNodeHelper::usage(void)const " (__imp_?usage@MapNodeHelper@Util@osgEarth@@QEBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) referenced in function "int __cdecl usage(char const *)" (?usage@@YAHPEBD@Z)
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl osgEarth::Util::PhongLightingGroup::PhongLightingGroup(void)" (__imp_??0PhongLightingGroup@Util@osgEarth@@QEAA@XZ) referenced in function main
osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl osgEarth::Util::PhongLightingGroup::~PhongLightingGroup(void)" (__imp_??1PhongLightingGroup@Util@osgEarth@@UEAA@XZ) referenced in function "public: virtual void * __cdecl osgEarth::Util::PhongLightingGroup::`scalar deleting destructor'(unsigned int)" (??_GPhongLightingGroup@Util@osgEarth@@UEAAPEAXI@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcConfigSubstitute referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcDefaultSubstitute referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcInitLoadConfigAndFonts referenced in function "struct _FcConfig * __cdecl getFontConfig(void)" (?getFontConfig@@YAPEAU_FcConfig@@XZ)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcFontMatch referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcNameParse referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcPatternDestroy referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcPatternGetString referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
src\applications\osgearth_viewer\osgearth_viewer.exe : fatal error LNK1120: 15 unresolved externals
UG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console && cd ."
[464/476] C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IF:\PRreview\vcpkg\buildtrees\osgearth\x64-windows-static-rel\build_include -IF:\PRreview\vcpkg\buildtrees\osgearth\src\gearth-3.7-7756cef1c2.clean\src -IF:\PRreview\vcpkg\installed\x64-windows-static\include /nologo /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /MP /MT /O2 /Oi /Gy /DNDEBUG /Z7  -std:c++14 -MT /showIncludes /Fosrc\osgEarthDrivers\engine_rex\CMakeFiles\osgdb_osgearth_engine_rex.dir\Unloader.cpp.obj /Fdsrc\osgEarthDrivers\engine_rex\CMakeFiles\osgdb_osgearth_engine_rex.dir\osgdb_osgearth_engine_rex.pdb /FS -c F:\PRreview\vcpkg\buildtrees\osgearth\src\gearth-3.7-7756cef1c2.clean\src\osgEarthDrivers\engine_rex\Unloader.cpp
FAILED: src/applications/osgearth_viewer/osgearth_viewer.exe 
C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=src\applications\osgearth_viewer\CMakeFiles\osgearth_viewer.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\osgearth_viewer.rsp  /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console && cd ."
ninja: build stopped: subcommand failed.
```

plevy committed:
new tag fixing win static linking

plevy committed:
update version

LilyWangLL commented:
All features passed with following triplets:
```
x64-windows
x64-windows-static
```
Usage test passed on `x64-windows`.

plevy committed:
Update ports/osgearth/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

plevy committed:
Update ports/osgearth/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

plevy committed:
update version

plevy commented:
> There are the following errors when installing `osgearth[tools]:x64-windows-static`, could you please resolve these errors?
> 
> ```
> [458/476] C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=src\applications\osgearth_viewer\CMakeFiles\osgearth_viewer.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\osgearth_viewer.rsp  /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBLINK: command "C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe @CMakeFiles\osgearth_viewer.rsp /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF /subsystem:console /MANIFEST:EMBED,ID=1" failed (exit code 1120) with the following output:
>    Creating library src\applications\osgearth_viewer\osgearth_viewer.lib and object src\applications\osgearth_viewer\osgearth_viewer.exp
> LINK : warning LNK4217: symbol '?initialize@osgEarth@@YAXAEAVArgumentParser@osg@@@Z (void __cdecl osgEarth::initialize(class osg::ArgumentParser &))' defined in 'osgEarth.lib(Registry.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
> LINK : warning LNK4217: symbol '??0UnitsType@osgEarth@@QEAA@PEBD0AEBW4Domain@Units@1@N@Z (public: __cdecl osgEarth::UnitsType::UnitsType(char const *,char const *,enum osgEarth::Units::Domain const &,double))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic initializer for 'BAM''(void)" (??__EBAM@Units@osgEarth@@YAXXZ)'
> LINK : warning LNK4217: symbol '??0UnitsType@osgEarth@@QEAA@PEBD0AEBV01@1@Z (public: __cdecl osgEarth::UnitsType::UnitsType(char const *,char const *,class osgEarth::UnitsType const &,class osgEarth::UnitsType const &))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic initializer for 'DATA_MILES_PER_HOUR''(void)" (??__EDATA_MILES_PER_HOUR@Units@osgEarth@@YAXXZ)'
> LINK : warning LNK4217: symbol '??1UnitsType@osgEarth@@QEAA@XZ (public: __cdecl osgEarth::UnitsType::~UnitsType(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function '"void __cdecl osgEarth::Units::`dynamic atexit destructor for 'BAM''(void)" (??__FBAM@Units@osgEarth@@YAXXZ)'
> LINK : warning LNK4217: symbol '??0ShaderGenerator@osgEarth@@QEAA@XZ (public: __cdecl osgEarth::ShaderGenerator::ShaderGenerator(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
> LINK : warning LNK4217: symbol '?run@ShaderGenerator@osgEarth@@QEAAXPEAVNode@osg@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAVStateSetCache@2@@Z (public: void __cdecl osgEarth::ShaderGenerator::run(class osg::Node *,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class osgEarth::StateSetCache *))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
> LINK : warning LNK4217: symbol '??_DShaderGenerator@osgEarth@@QEAAXXZ (public: void __cdecl osgEarth::ShaderGenerator::`vbase destructor'(void))' defined in 'osgEarth.lib(ShaderGenerator.cpp.obj)' is imported by 'osgearth_viewer.cpp.obj' in function 'main'
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static class osgEarth::MapNode * __cdecl osgEarth::MapNode::get(class osg::ref_ptr<class osg::Node> &,unsigned int)" (__imp_?get@MapNode@osgEarth@@SAPEAV12@AEAV?$ref_ptr@VNode@osg@@@osg@@I@Z) referenced in function main
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl osgEarth::Util::EarthManipulator::EarthManipulator(class osg::ArgumentParser &)" (__imp_??0EarthManipulator@Util@osgEarth@@QEAA@AEAVArgumentParser@osg@@@Z) referenced in function main
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual char const * __cdecl osgEarth::Util::EarthManipulator::className(void)const " (__imp_?className@EarthManipulator@Util@osgEarth@@UEBAPEBDXZ) referenced in function "[thunk]:public: virtual char const * __cdecl osgEarth::Util::EarthManipulator::className`vtordisp{4294967292,0}' (void)const " (?className@EarthManipulator@Util@osgEarth@@$4PPPPPPPM@A@EBAPEBDXZ)
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) protected: void __cdecl osgEarth::Util::EarthManipulator::`vbase destructor'(void)" (__imp_??_DEarthManipulator@Util@osgEarth@@IEAAXXZ) referenced in function "protected: virtual void * __cdecl osgEarth::Util::EarthManipulator::`scalar deleting destructor'(unsigned int)" (??_GEarthManipulator@Util@osgEarth@@MEAAPEAXI@Z)
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class osg::ref_ptr<class osg::Node> __cdecl osgEarth::Util::MapNodeHelper::load(class osg::ArgumentParser &,class osgViewer::ViewerBase *)const " (__imp_?load@MapNodeHelper@Util@osgEarth@@QEBA?AV?$ref_ptr@VNode@osg@@@osg@@AEAVArgumentParser@5@PEAVViewerBase@osgViewer@@@Z) referenced in function main
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgEarth::Util::MapNodeHelper::usage(void)const " (__imp_?usage@MapNodeHelper@Util@osgEarth@@QEBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) referenced in function "int __cdecl usage(char const *)" (?usage@@YAHPEBD@Z)
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl osgEarth::Util::PhongLightingGroup::PhongLightingGroup(void)" (__imp_??0PhongLightingGroup@Util@osgEarth@@QEAA@XZ) referenced in function main
> osgearth_viewer.cpp.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: virtual __cdecl osgEarth::Util::PhongLightingGroup::~PhongLightingGroup(void)" (__imp_??1PhongLightingGroup@Util@osgEarth@@UEAA@XZ) referenced in function "public: virtual void * __cdecl osgEarth::Util::PhongLightingGroup::`scalar deleting destructor'(unsigned int)" (??_GPhongLightingGroup@Util@osgEarth@@UEAAPEAXI@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcConfigSubstitute referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcDefaultSubstitute referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcInitLoadConfigAndFonts referenced in function "struct _FcConfig * __cdecl getFontConfig(void)" (?getFontConfig@@YAPEAU_FcConfig@@XZ)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcFontMatch referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcNameParse referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcPatternDestroy referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> osgText.lib(Font.cpp.obj) : error LNK2019: unresolved external symbol FcPatternGetString referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl osgText::findFontFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?findFontFile@osgText@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV23@@Z)
> src\applications\osgearth_viewer\osgearth_viewer.exe : fatal error LNK1120: 15 unresolved externals
> UG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console && cd ."
> [464/476] C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\cl.exe   /TP -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -IF:\PRreview\vcpkg\buildtrees\osgearth\x64-windows-static-rel\build_include -IF:\PRreview\vcpkg\buildtrees\osgearth\src\gearth-3.7-7756cef1c2.clean\src -IF:\PRreview\vcpkg\installed\x64-windows-static\include /nologo /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /MP /MT /O2 /Oi /Gy /DNDEBUG /Z7  -std:c++14 -MT /showIncludes /Fosrc\osgEarthDrivers\engine_rex\CMakeFiles\osgdb_osgearth_engine_rex.dir\Unloader.cpp.obj /Fdsrc\osgEarthDrivers\engine_rex\CMakeFiles\osgdb_osgearth_engine_rex.dir\osgdb_osgearth_engine_rex.pdb /FS -c F:\PRreview\vcpkg\buildtrees\osgearth\src\gearth-3.7-7756cef1c2.clean\src\osgEarthDrivers\engine_rex\Unloader.cpp
> FAILED: src/applications/osgearth_viewer/osgearth_viewer.exe 
> C:\Windows\system32\cmd.exe /C "cd . && F:\PRreview\vcpkg\downloads\tools\cmake-3.30.1-windows\cmake-3.30.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=src\applications\osgearth_viewer\CMakeFiles\osgearth_viewer.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~3\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\link.exe  @CMakeFiles\osgearth_viewer.rsp  /out:src\applications\osgearth_viewer\osgearth_viewer.exe /implib:src\applications\osgearth_viewer\osgearth_viewer.lib /pdb:src\applications\osgearth_viewer\osgearth_viewer.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console && cd ."
> ninja: build stopped: subcommand failed.
> ```

I believe we have resolved these issues in the library and updated the release.

dg0yt commented on ports/osgearth/portfile.cmake:
```diff
@@ -2,7 +2,7 @@ vcpkg_from_github(
     REPO gwaldron/osgearth
     REF "osgearth-${VERSION}"
-    SHA512 fe80e8ce40669a896e046a829ba2a9a7dc3d79e9508362e71149c51078eb44486532c850ec00df88808540c090336a99275fd4496be125131b0774371792aed0
+    SHA512 2f764eb1fff21cff57a04eceb73643b372c8b70899114c88cdf9928d525517214959745cb70b99ad3d1def946bfb3f58f16b4969ee4117b7563398f2410ee3e2
```
BTW please never change an exisiting tag / release.

BillyONeal commented:
Thanks for the update!

@dg0yt Thanks for the review!
